src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/model/identifiers/CROIdentifier.kt

```diff
@@ -0,0 +1,93 @@
+package uk.gov.justice.digital.hmpps.personrecord.model.identifiers
+
+class CROIdentifier(inputCroId: String, inputFingerprint: Boolean) {
+
+  val croId: String = inputCroId
+  val fingerprint: Boolean = inputFingerprint
+
+  val valid: Boolean
+    get() = croId.isNotEmpty()
+
+  override fun equals(other: Any?): Boolean {
+    if (this === other) return true
+    if (javaClass != other?.javaClass) return false
+    other as CROIdentifier
+    return croId == other.croId
```

Comment:
Although it is unlikely to happen in practice, this check opens up the possibility of two CROIdentifiers with the same corId and a different fingerprint being equal. Equals checks must compare all relevant fields.

```  override fun equals(other: Any?): Boolean {
    return EqualsBuilder.reflectionEquals(this, other)
  }```
  
  
  This is a failsafe way to implement equals - there is also a HashcodeBuilder which works in the same way
  
  ```https://mvnrepository.com/artifact/org.apache.commons/commons-lang3/3.14.0``` is the dependency

Reply:
Added 👍